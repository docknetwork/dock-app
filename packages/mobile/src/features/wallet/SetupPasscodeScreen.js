import React from 'react';
import {
  Button,
  Header,
  Footer,
  Content,
  ScreenContainer,
  Typography,
} from '../../design-system';
import {BackButton} from '../../design-system/buttons';
import LockClosedIcon from '../../assets/icons/lock-closed.svg';
import {translate} from 'src/locales';

export function SetupPasscodeScreen() {
  return (
    <ScreenContainer testID="create-wallet-screen">
      <Header>
        <BackButton />
      </Header>
      <Content marginLeft={26} marginRight={26}>
        <Typography variant="h1" marginTop={52}>
          {translate('setup_passcode.title')}
        </Typography>
        <Typography marginTop={12}>
          {translate('setup_passcode.description')}
        </Typography>
      </Content>
      <Footer marginBottom={114} marginLeft={26} marginRight={26}>
        <Button full icon={<LockClosedIcon />} testID="create-wallet-btn">
          {translate('setup_passcoe.submit')}
        </Button>
      </Footer>
    </ScreenContainer>
  );
}
