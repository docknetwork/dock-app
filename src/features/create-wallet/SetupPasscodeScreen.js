import React from 'react';
import {
  Button,
  Header,
  Footer,
  Content,
  ScreenContainer,
  Typography,
} from '../../design-system';
import {BackButton} from '../../design-system/buttons';
import LockClosedIcon from '../../assets/icons/lock-closed.svg';

export function SetupPasscodeScreen() {
  return (
    <ScreenContainer testID="create-wallet-screen">
      <Header>
        <BackButton />
      </Header>
      <Content marginLeft={26} marginRight={26}>
        <Typography
          fontFamily="Montserrat"
          fontSize={24}
          lineHeight={32}
          fontWeight="600"
          color="#fff"
          marginTop={52}>
          Setup Passcode
        </Typography>
        <Typography
          fontSize={16}
          lineHeight={24}
          fontWeight="400"
          marginTop={12}>
          Add a passcode to keep your wallet safe
        </Typography>
      </Content>
      <Footer marginBottom={114} marginLeft={26} marginRight={26}>
        <Button full icon={<LockClosedIcon />} testID="create-wallet-btn">
          Setup Passcode
        </Button>
      </Footer>
    </ScreenContainer>
  );
}
