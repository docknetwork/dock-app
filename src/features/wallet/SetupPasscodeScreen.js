import React from 'react';
import {
  Button,
  Header,
  Footer,
  Content,
  ScreenContainer,
  Typography,
} from '../../design-system';
import {BackButton} from '../../design-system/buttons';
import LockClosedIcon from '../../assets/icons/lock-closed.svg';
import {translate} from 'src/locales';
import {navigate} from 'src/core/navigation';
import {Routes} from 'src/core/routes';

export function SetupPasscodeScreen() {
  return (
    <ScreenContainer testID="setupPasscodeScreen">
      <Header>
        <BackButton />
      </Header>
      <Content marginLeft={26} marginRight={26}>
        <Typography variant="h1" marginTop={52}>
          {translate('setup_passcode.title')}
        </Typography>
        <Typography marginTop={12}>
          {translate('setup_passcode.description')}
        </Typography>
      </Content>
      <Footer marginBottom={114} marginLeft={26} marginRight={26}>
        <Button
          full
          icon={<LockClosedIcon />}
          testID="create-wallet-btn"
          onPress={() => {
            navigate(Routes.CREATE_WALLET_PASSCODE);
          }}>
          {translate('setup_passcode.submit')}
        </Button>
      </Footer>
    </ScreenContainer>
  );
}
