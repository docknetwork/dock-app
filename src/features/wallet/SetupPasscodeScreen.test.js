import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {SetupPasscodeScreen} from './SetupPasscodeScreen';

const mockStore = configureMockStore();

describe('SetupPasscodeScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<SetupPasscodeScreen />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
