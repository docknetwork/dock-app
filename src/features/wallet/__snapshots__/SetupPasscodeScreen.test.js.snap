// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupPasscodeScreen should render correctly 1`] = `
<RNCSafeAreaView
  accessibilityID="setupPasscodeScreen"
  style={
    Object {
      "backgroundColor": "#18181B",
      "flex": 1,
    }
  }
  testID="setupPasscodeScreen"
>
  <AppGlobalHeader />
  <Styled(View)>
    <BackButton
      accessibilityID="BackButton"
      testID="BackButton"
    />
  </Styled(View)>
  <Styled(ScrollView)
    marginLeft={26}
    marginRight={26}
  >
    <Styled(Component)
      marginTop={52}
      variant="h1"
    >
      Setup Passcode
    </Styled(Component)>
    <Styled(Component)
      marginTop={12}
    >
      Add a passcode to keep your wallet safe
    </Styled(Component)>
  </Styled(ScrollView)>
  <Styled(Box)
    marginBottom={114}
    marginLeft={26}
    marginRight={26}
  >
    <Button
      accessibilityLabel="create-wallet-btn"
      full={true}
      icon={<SvgFile />}
      onPress={[Function]}
      testID="create-wallet-btn"
    >
      Setup Passcode
    </Button>
  </Styled(Box)>
</RNCSafeAreaView>
`;
